TestVectors/dafny/TestVectors/src/WriteVectors.dfy

```diff
@@ -154,4 +161,120 @@ module {:options "-functionSyntax:4"} WriteVectors {
     case 5 => Success(AllEsdkV4NoReqEc.Tests + AllEsdkV4WithReqEc.Tests)
     case _ => Failure("Only version 4 of generate manifest is supported\n")
   }
+
+  predicate TestVectorListHasNoDuplicates(xs: seq<EsdkTestVectors.EsdkEncryptTestVector>)
+  {
+    forall i, j :: 0 <= i < j < |xs| ==> xs[i].description != xs[j].description
+  }
+
+  predicate DescriptionLessThan(x: EsdkTestVectors.EsdkEncryptTestVector, y: EsdkTestVectors.EsdkEncryptTestVector) {
+    Below(x.description, y.description)
+  }
```

Comment:
Likely too much work ATM, but we should come back and change this to:

```suggestion
  predicate DescriptionLessThan(x: EsdkTestVectors.EsdkEncryptTestVector, y: EsdkTestVectors.EsdkEncryptTestVector) {
    && Below(x.description, y.description)
    && Below(x.algorithmSuiteId, y.algorithmSuiteId)
  }
```

I know that this might be complicated so I won't block on it.